assignments/prototypes.js

```diff
@@ -102,9 +132,21 @@
   console.log(archer.greet()); // Lilith offers a greeting in Elvish.
   console.log(mage.takeDamage()); // Bruce took damage.
   console.log(swordsman.destroy()); // Sir Mustachio was removed from the game.
-*/
+
 
   // Stretch task: 
   // * Create Villain and Hero constructor functions that inherit from the Humanoid constructor function.  
+function Villain(attributes){
+  Humanoid.call(this, attributes);
+  this.isEvil = attributes.isEvil
+}
+Villain.prototype = Object.create(Humanoid.prototype)
+
+function Hero(attributes){
+  Humanoid.call(this, attributes);
+  this.isGood = attributes.isGood
+}
+Hero.prototype = Object.create(Humanoid.prototype)
+
   // * Give the Hero and Villains different methods that could be used to remove health points from objects which could result in destruction if health gets to 0 or drops below 0;
```

Comment:
Great job on stretch.  Here is how I did it:

// New Villian Object Inherits from Humanoid
function Villian(villianAttributes) {
    Humanoid.call(this, villianAttributes);
}

Villian.prototype = Object.create(Humanoid.prototype);

Villian.prototype.mindSpell = function() {
    return `${this.name} just cast a spell on your mind causing you to lose 5 health points.`
}

// New Hero Object Inherits from Humanoid
function Hero(heroAttributes) {
  Humanoid.call(this, heroAttributes);
}

Hero.prototype = Object.create(Humanoid.prototype);

Hero.prototype.counterSpell = function() {
  return `${this.name} just cast a counter spell on your mind causing you to lose 7 health points.`
}

const badman = new Villian({
      createdAt: new Date(),
      dimensions: {
          length: 1,
          width: 2,
          height: 4,
      },
      hp: 15,
      name: 'Shoeface',
      faction: 'House of Evil',
      weapons: [
          'Spells',
          'Wand',
      ],
      language: 'Orcish',
  });

  const goodguy = new Hero({
      createdAt: new Date(),
      dimensions: {
          length: 2,
          width: 1,
          height: 2,
      },
      hp: 15,
      name: 'Aelforn',
      faction: 'People of the West',
      weapons: [
          'Spells',
          'Magic Sword',
      ],
      language: 'Elvish',
  });

console.log(badman)
  console.log(badman.mindSpell());
  goodguy.hp -= 5;
  console.log(goodguy.hp);
  console.log(goodguy);
  console.log(goodguy.counterSpell());
  badman.hp -= 7;
  console.log(badman.hp);
  console.log(badman.mindSpell());
  goodguy.hp -= 5;
  console.log(goodguy.hp);
  console.log(goodguy.counterSpell());
  badman.hp -= 7;
  console.log(badman.hp);
  console.log(badman.mindSpell());
  goodguy.hp -= 5;
  console.log(goodguy.hp);
  console.log(goodguy.destroy());